### ISSUE TITLE
Error when injecting and starting workspace agent

### ISSUE THREAD

**Issue by jedyn96** [eclipse-che/che#940]
[STDOUT] Listening for transport dt_socket at address: 4403
[STDOUT] 2016-03-30 17:48:27,040[main] [INFO] [o.a.c.s.VersionLoggerListener 89] - Server version: Apache Tomcat/8.0.29
[STDOUT] 2016-03-30 17:48:27,044[main] [INFO] [o.a.c.s.VersionLoggerListener 91] - Server built: Nov 20 2015 09:18:00 UTC
[STDOUT] 2016-03-30 17:48:27,052[main] [INFO] [o.a.c.s.VersionLoggerListener 93] - Server number: 8.0.29.0
[STDOUT] 2016-03-30 17:48:27,053[main] [INFO] [o.a.c.s.VersionLoggerListener 95] - OS Name: Linux
[STDOUT] 2016-03-30 17:48:27,053[main] [INFO] [o.a.c.s.VersionLoggerListener 97] - OS Version: 4.2.0-34-generic
[STDOUT] 2016-03-30 17:48:27,054[main] [INFO] [o.a.c.s.VersionLoggerListener 99] - Architecture: amd64
[STDOUT] 2016-03-30 17:48:27,056[main] [INFO] [o.a.c.s.VersionLoggerListener 101] - Java Home: /opt/jdk1.8.0_45/jre
[STDOUT] 2016-03-30 17:48:27,056[main] [INFO] [o.a.c.s.VersionLoggerListener 103] - JVM Version: 1.8.0_45-b14
[STDOUT] 2016-03-30 17:48:27,057[main] [INFO] [o.a.c.s.VersionLoggerListener 105] - JVM Vendor: Oracle Corporation
[STDOUT] 2016-03-30 17:48:27,058[main] [INFO] [o.a.c.s.VersionLoggerListener 107] - CATALINA_BASE: /home/user/che/ws-agent
[STDOUT] 2016-03-30 17:48:27,059[main] [INFO] [o.a.c.s.VersionLoggerListener 109] - CATALINA_HOME: /home/user/che/ws-agent
[STDOUT] 2016-03-30 17:48:27,066[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.util.logging.config.file=/home/user/che/ws-agent/conf/logging.properties
[STDOUT] 2016-03-30 17:48:27,068[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
[STDOUT] 2016-03-30 17:48:27,071[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Xms256m
[STDOUT] 2016-03-30 17:48:27,072[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Xmx1024m
[STDOUT] 2016-03-30 17:48:27,072[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.security.egd=file:/dev/./urandom
[STDOUT] 2016-03-30 17:48:27,073[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dche.logs.dir=/home/user/che/ws-agent/logs
[STDOUT] 2016-03-30 17:48:27,073[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcom.sun.management.jmxremote
[STDOUT] 2016-03-30 17:48:27,074[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcom.sun.management.jmxremote.ssl=false
[STDOUT] 2016-03-30 17:48:27,074[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcom.sun.management.jmxremote.authenticate=false
[STDOUT] 2016-03-30 17:48:27,075[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dche.local.conf.dir=/mnt/che/conf
[STDOUT] 2016-03-30 17:48:27,075[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -agentlib:jdwp=transport=dt_socket,address=4403,server=y,suspend=n
[STDOUT] 2016-03-30 17:48:27,076[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.endorsed.dirs=/home/user/che/ws-agent/endorsed
[STDOUT] 2016-03-30 17:48:27,077[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcatalina.base=/home/user/che/ws-agent
[STDOUT] 2016-03-30 17:48:27,078[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Dcatalina.home=/home/user/che/ws-agent
[STDOUT] 2016-03-30 17:48:27,079[main] [INFO] [o.a.c.s.VersionLoggerListener 115] - Command line argument: -Djava.io.tmpdir=/home/user/che/ws-agent/temp
[STDOUT] 2016-03-30 17:48:27,674[main] [INFO] [o.a.c.http11.Http11NioProtocol 425] - Initializing ProtocolHandler ["http-nio-4401"]
[STDOUT] 2016-03-30 17:48:27,720[main] [INFO] [o.a.t.util.net.NioSelectorPool 72] - Using a shared selector for servlet write/read
[STDOUT] 2016-03-30 17:48:27,725[main] [INFO] [o.a.catalina.startup.Catalina 587] - Initialization processed in 1204 ms
[STDOUT] 2016-03-30 17:48:27,789[main] [INFO] [c.m.JmxRemoteLifecycleListener 332] - The JMX Remote Listener has configured the registry on port 32002 and the server on port 32102 for the Platform server
[STDOUT] 2016-03-30 17:48:27,789[main] [INFO] [o.a.c.core.StandardService 433] - Starting service Catalina
[STDOUT] 2016-03-30 17:48:27,790[main] [INFO] [o.a.c.core.StandardEngine 259] - Starting Servlet Engine: Apache Tomcat/8.0.29
[STDOUT] 2016-03-30 17:48:27,889[ost-startStop-1] [INFO] [o.a.c.startup.HostConfig 916] - Deploying web application archive /home/user/che/ws-agent/webapps/ide.war
[STDOUT] Wed Mar 30 17:48:39 UTC 2016 - [localhost-startStop-1] Product-specified preferences called before plugin is started
[STDOUT] 2016-03-30 17:48:40,144[rrest.WSClient1] [INFO] [.c.a.c.n.WSocketEventBusClient 223] - Open connection to ws://che-host:8080/ide/api/eventbus/?token=dummy_token. 
[STDOUT] 2016-03-30 17:48:40,760[ost-startStop-1] [INFO] [o.a.c.startup.HostConfig 980] - Deployment of web application archive /home/user/che/ws-agent/webapps/ide.war has finished in 12,691 ms
[STDOUT] 2016-03-30 17:48:40,789[main] [INFO] [o.a.c.http11.Http11NioProtocol 470] - Starting ProtocolHandler ["http-nio-4401"]
[STDOUT] 2016-03-30 17:48:40,795[main] [INFO] [o.a.catalina.startup.Catalina 638] - Server startup in 13069 ms
[STDOUT] 2016-03-30 17:48:41,323[nio-4401-exec-3] [WARN ] [i.ContainerResponse$BodyWriter 90] - Client has aborted connection. Response writing omitted.
[STDOUT] 2016-03-30 17:48:41,323[nio-4401-exec-2] [WARN ] [i.ContainerResponse$BodyWriter 90] - Client has aborted connection. Response writing omitted.
[STDOUT] 2016-03-30 17:48:41,323[nio-4401-exec-1] [WARN ] [i.ContainerResponse$BodyWriter 90] - Client has aborted connection. Response writing omitted.
[STDOUT] 2016-03-30 17:48:41,869[nio-4401-exec-7] [ERROR] [c.s.o.RemoteOAuthTokenProvider 75] - Exception on token retrieval, message : Unsupported OAuth provider github

-->Error when injecting and starting workspace agent on ubuntu 14.04.

-->maven version:-
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T22:11:47+05:30)
Maven home: /usr/local/apache-maven
Java version: 1.8.0_77, vendor: Oracle Corporation
Java home: /usr/local/java-current/jre
Default locale: en_IN, platform encoding: UTF-8
OS name: "linux", version: "4.2.0-34-generic", arch: "amd64", family: "unix"

-->Java version:-
java version "1.8.0_77"
Java(TM) SE Runtime Environment (build 1.8.0_77-b03)
Java HotSpot(TM) 64-Bit Server VM (build 25.77-b03, mixed mode)

-->git version:-
git version 1.9.1

The dashboard shows this error message dialog box :- 
"Error while creating the project
Internal Server Error: fatal: unable to access 'https://github.com/che-samples/console-java-simple.git/': Could not resolve host: github.com"

**Comment by ghost** [comment]
The simplest way to check if it is a Che or a connectivity issue is to manually run:

`docker run -ti codenvy/ubuntu_jdk8 bash`

When you get a shell in the container:

`cd  /home/user`
`git clone https://github.com/che-samples/console-java-simple.git`

The error says it all - for some reason Git cannot resolve hostname `github.com`. The workspace agent itself has started ok. You may want to try to create an empty workspace and open it in IDE.

**Comment by jedyn96** [comment]
i tried in container and it didn't resolve github.com, same error. Can you help me with this? I have no idea how to solve this

**Comment by ghost** [comment]
Proxy maybe? At this point Che is not involved. The problem is that there's no connection from within a container.

Restarting Docker service is a good idea too.

**Comment by jedyn96** [comment]
Hello eivantsov, I was able to use proper and dns servers and so the above test worked :- 

umang@umang-HP-Pavilion-g6-Notebook-PC:~$ docker run -ti codenvy/ubuntu_jdk8 bash
user@52dd8ac3d2bd:/$ cd /home/user
user@52dd8ac3d2bd:~$ git clone https://github.com/che-samples/console-java-simple.git
Cloning into 'console-java-simple'...
remote: Counting objects: 88, done.
remote: Total 88 (delta 0), reused 0 (delta 0), pack-reused 88
Unpacking objects: 100% (88/88), done.
Checking connectivity... done.

So , it is connecting to github and also now i am able to open the IDE but I can't create projects from within IDE from workspace tab. The create button is disabled.

![image](https://cloud.githubusercontent.com/assets/11131874/14219151/3f5f5740-f876-11e5-9de4-4f1b8cede4f6.png)

Also I can import projects from github. I tried hello-world, build it and ran it correctly
.
